avocado/core/requirements/resolver.py

```diff
@@ -0,0 +1,30 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2021
+# Authors: Willian Rampazzo <willianr@redhat.com>
+
+from ..nrunner import Runnable
+
+
+class RequirementsResolver:
+
+    name = 'requirements'
+    description = 'Requirements resolver for tests with requirements'
+
+    @staticmethod
+    def resolve(runnable):
+        requirements_runnables = []
+        for requirement in runnable.requirements:
+            kind = 'requirement-%s' % requirement.get('type')
+            requirement_runnable = Runnable(kind, None, **requirement)
```

Comment:
I think it's wrong because we're passing unintended data as `kwargs`, knowingly and on purpose.  The `type` key and its value in the requirement were never intended to be used that way -- it's to be consumed by the requirement resolver, and not by the runner.

What if, besides `type` we decide that we'll also let users decide the `priority` of a requirement, or the `condition` in which it should be fulfilled? Those are probably within the scope of the resolver, and the runners would never have to deal with them.  Given that example, should we keep passing all of those as `kwargs` to the runners? IMO, the answer is a firm no.